ext/gmp/gmp.c

```diff
@@ -1164,34 +1188,24 @@ ZEND_FUNCTION(gmp_abs)
 ZEND_FUNCTION(gmp_fact)
 {
 	zval *a_arg;
+	mpz_ptr gmpnum;
+	mpz_ptr gmpnum_result;
+	gmp_temp_t temp_a;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &a_arg) == FAILURE){
 		RETURN_THROWS();
 	}
 
-	if (IS_GMP(a_arg)) {
-		mpz_ptr gmpnum_tmp = GET_GMP_FROM_ZVAL(a_arg);
-		if (mpz_sgn(gmpnum_tmp) < 0) {
-			php_error_docref(NULL, E_WARNING, "Number has to be greater than or equal to 0");
-			RETURN_FALSE;
-		}
-	} else {
-		/* Use convert_to_number first to detect getting non-integer */
-		convert_scalar_to_number(a_arg);
-		if (Z_TYPE_P(a_arg) != IS_LONG) {
-			convert_to_long(a_arg);
-			if (Z_LVAL_P(a_arg) >= 0) {
-				/* Only warn if we'll make it past the non-negative check */
-				php_error_docref(NULL, E_WARNING, "Number has to be an integer");
-			}
-		}
-		if (Z_LVAL_P(a_arg) < 0) {
-			php_error_docref(NULL, E_WARNING, "Number has to be greater than or equal to 0");
-			RETURN_FALSE;
-		}
+	FETCH_GMP_ZVAL(gmpnum, a_arg, temp_a, 1);
+	FREE_GMP_TEMP(temp_a);
```

Comment:
Shouldn't this function just accept `int` instead? Makes little sense to convert to GMP first and then convert back to integer.

Reply:
Well the function should be able to accept a GMP object, but it's true that this code makes little sense atm.